app/javascript/mastodon/features/onboarding/profile.jsx

```diff
@@ -0,0 +1,162 @@
+          <div className='onboarding__profile'>
+            <label className={classNames('app-form__header-input', { selected: !!headerPreview })} title={intl.formatMessage(messages.uploadHeader)}>
+              <input
+                type='file'
+                hidden
+                ref={headerFileRef}
+                accept='image/*'
+                onChange={handleHeaderChange}
+              />
+
+              {headerPreview && <img src={headerPreview} alt='' />}
+
+              <Icon icon={headerPreview ? EditIcon : AddPhotoAlternateIcon} />
+            </label>
+
+            <label className={classNames('app-form__avatar-input', { selected: !!avatarPreview })} title={intl.formatMessage(messages.uploadAvatar)}>
+              <input
+                type='file'
+                hidden
+                ref={avatarFileRef}
+                accept='image/*'
+                onChange={handleAvatarChange}
+              />
+
+              {avatarPreview && <img src={avatarPreview} alt='' />}
+
+              <Icon icon={avatarPreview ? EditIcon : AddPhotoAlternateIcon} />
+            </label>
+          </div>
+
+          <div className='input with_block_label'>
+            <label><FormattedMessage id='onboarding.profile.display_name' defaultMessage='Display name' /></label>
+            <span className='hint'><FormattedMessage id='onboarding.profile.display_name_hint' defaultMessage='Your full name or your fun name... ' /></span>
+            <div className='label_input'>
+              <input type='text' value={displayName} onChange={handleDisplayNameChange} maxLength={30} />
+            </div>
+          </div>
+
+          <div className='input with_block_label'>
+            <label><FormattedMessage id='onboarding.profile.note' defaultMessage='Bio' /></label>
+            <span className='hint'><FormattedMessage id='onboarding.profile.note_hint' defaultMessage='You can @mention other people or #hashtags...' /></span>
+            <div className='label_input'>
+              <textarea value={note} onChange={handleNoteChange} />
+            </div>
+          </div>
+        </div>
+
+        <label className='report-dialog-modal__toggle'>
+          <Toggle checked={discoverable} onChange={handleDiscoverableChange} />
+          <FormattedMessage id='onboarding.profile.discoverable' defaultMessage='Feature profile and posts in discovery algorithms' />
+        </label>
+
+        <label className='report-dialog-modal__toggle'>
+          <Toggle checked={indexable} onChange={handleIndexableChange} />
+          <FormattedMessage id='onboarding.profile.indexable' defaultMessage='Include public posts in search results' />
+        </label>
+
+        <div className='onboarding__footer'>
+          <Button block onClick={handleSubmit} disabled={isSaving}>{isSaving ? <LoadingIndicator /> : <FormattedMessage id='onboarding.profile.save_and_continue' defaultMessage='Save and continue' />}</Button>
```

Comment:
It might be a good idea to add a `loading: boolean` props on the `<Button>` component, as this is a pattern we can use in other places